packages/financial-templates-lib/src/logger/PagerDutyTransport.ts

```diff
@@ -3,22 +3,36 @@
 // notification is not acknowledged by the person on call within 30 mins a second person is contacted until the warning
 // is acknowledged. If set to `high` the incident is aggressively escalated. If no acknowledgement within 5 minutes a
 //  second person is contacted until the message is acknowledged.
-
-const Transport = require("winston-transport");
-const pdClient = require("node-pagerduty");
-
-module.exports = class PagerDutyTransport extends Transport {
-  constructor(winstonOpts, pagerDutyOptions) {
+import Transport from "winston-transport";
+import PagerDutyClient from "node-pagerduty";
+
+type TransportOptions = ConstructorParameters<typeof Transport>[0];
+
+export class PagerDutyTransport extends Transport {
+  private readonly pd: PagerDutyClient;
+  private readonly fromEmail: string;
+  private readonly defaultServiceId: string;
+  private readonly customServices: { [key: string]: string };
+  constructor(
+    winstonOpts: TransportOptions,
+    {
+      pdApiToken,
+      fromEmail,
+      defaultServiceId,
+      customServices = {},
+    }: { pdApiToken: string; fromEmail: string; defaultServiceId: string; customServices: { [key: string]: string } }
+  ) {
     super(winstonOpts);
 
-    this.pd = new pdClient(pagerDutyOptions.pdApiToken);
+    this.pd = new PagerDutyClient(pdApiToken);
 
-    this.fromEmail = pagerDutyOptions.fromEmail;
-    this.defaultServiceId = pagerDutyOptions.defaultServiceId;
-    this.customServices = pagerDutyOptions.customServices || {};
+    this.fromEmail = fromEmail;
+    this.defaultServiceId = defaultServiceId;
+    this.customServices = customServices;
```

Comment:
will this still work fine if `customServices` is empty? in that case, we need it to default to a blank object like `{}`. I think it'll do this from your defaults in the constructor but need to double-check as this is key to this transport still working in the case the user only uses the `defaultServiceId`

Reply:
looking at it again, I agree. the defaults make this work fine.